utils.js

```diff
@@ -1,4 +1,48 @@
-const {forEach, map} = require('wu');
+const {flatten, forEach, map} = require('wu');
+
+
+function identity(x) {
+    return x;
+}
+
+
+// From an iterable return the best item, according to an arbitrary comparator
+// function. In case of a tie, the first item wins.
+function best(iterable, by, isBetter) {
+    let bestSoFar, bestKeySoFar;
+    let isFirst = true;
+    forEach(
+        function (item) {
+            const key = by(item);
+            if (isBetter(key, bestKeySoFar) || isFirst) {
+                bestSoFar = item;
+                bestKeySoFar = key;
+                isFirst = false;
+            }
+        },
+        iterable);
+    if (isFirst) {
+        throw 'Tried to call best() on empty iterable';
```

Comment:
⚠️ Probably an uber-nit, but not sure if we want to make sure we throw exceptions instead of string literals.

Ref: http://eslint.org/docs/rules/no-throw-literal

> _"The fundamental benefit of Error objects is that they automatically keep track of where they were built and originated."_